### DISCUSSION TITLE
Help with Setting Subdomain for Admin Area

### DISCUSSION THREAD

**Discussion by gentium-maroc** [vercel/community#6522] [Help]
### Summary

Hi, I have my main domain called gentium.io, and it's currently pointed to a project on Vercel (a landing page). Additionally, I have another project on Vercel, which serves as my admin area. I want to use app.gentium.io (adding 'app.') for this admin area. How can I accomplish that, please?

### Steps to Reproduce

Hi, I have my main domain called gentium.io, and it's currently pointed to a project on Vercel (a landing page). Additionally, I have another project on Vercel, which serves as my admin area. I want to use app.gentium.io (adding 'app.') for this admin area. How can I accomplish that, please?

**Comment by versecafe** [comment]
add `*.gentium.io` to the allowed domains and then in `middleware.ts` *assuming you're using NextJS*
```ts
if (hostname == `app.${process.env.NEXT_PUBLIC_ROOT_DOMAIN}`) {
    return NextResponse.rewrite(
      new URL(`/app${path === "/" ? "" : path}`, req.url),
    );
  }
if (hostname == `admin.${process.env.NEXT_PUBLIC_ROOT_DOMAIN}`) {
    return NextResponse.rewrite(
      new URL(`/admin${path === "/" ? "" : path}`, req.url),
    );
  }
  ```
assuming a file system like
```
/app, app router folder
  /admin
  /app, routes by default to domain.com/app rewrites to app.domain.com
  /home
```
